src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sideOfTriangle = [a, b, c].sort((a1, b1) => b1 - a1);
+    const condition =
+      sideOfTriangle[0] >= sideOfTriangle[1] + sideOfTriangle[2];
+
+    if (a <= 0 || b <= 0 || c <= 0 || condition) {
+      throw new Error(
+        'Invalid triangle sides: ensure all sides are possitive.',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = (1 / 2) * (this.a + this.b + this.c);
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid radius: radius must be a positive number.');
+    }
```

Comment:
```suggestion
   const circleErrorMessage = 'Invalid radius: radius must be a positive number.';
   
   checkOnError(circleErrorMessage, radius)
```
Also the message itself can also be put into a variable

Reply:
Thank you for review! I will definitely change the code based on your remarks:)